autoload/ale/fixers/prettier.vim

```diff
@@ -10,6 +10,10 @@ call ale#Set('javascript_prettier_options', '')
 function! s:FindConfig(buffer) abort
     for l:filename in [
     \   '.prettierrc',
+    \   '.prettierrc.yaml',
+    \   '.prettierrc.yml',
+    \   '.prettierrc.json',
+    \   '.prettierrc.js',
     \   'prettier.config.js',
     \   'package.json',
     \ ]
```

Comment:
We should follow the same precedence that cosmiconfig uses, which is mentioned [here](https://github.com/davidtheclark/cosmiconfig#rcextensions) and implemented [here](https://github.com/davidtheclark/cosmiconfig/blob/master/src/loadRc.js).

I reckon the correct order is.

1. No extension
2. `.json`
3. `.yaml`
4. `.yml`
5. `.js`
6. `.config.js`
7. `package.json`

Rather confusingly, ESLint has a similar set of configuration files... with a completed different order of precedence. https://eslint.org/docs/user-guide/configuring#configuration-file-formats

Reply:
Ahh, that is true! Updated the commit to reflect this list.

It turns out that `package.json` is being matched first, but ONLY if it has a `"prettier": {}` element inside it. I think this is an acceptable soultion, even though the best would have been to use the command `prettier --find-config-path [filename]` to get the config directly from prettier.